test/AutoRest.TestServer.Tests/Mgmt/TestProjects/TestProjectTests.cs

```diff
@@ -53,6 +52,51 @@ public void ValidateMethodExists(string methodName)
             }
         }
 
+        [Ignore("TODO ADO item 1111")]
+        [Test]
+        public void ValidateResourceGroupExtensions()
+        {
+            if (_projectName.Equals(""))
+            {
+                return;
+            }
+            Type[] allTypes = Assembly.GetExecutingAssembly().GetTypes();
+
+            Type resourceExtensions = allTypes.FirstOrDefault(t => t.Name == "ResourceGroupExtensions" && t.Namespace == _projectName);
+            Assert.NotNull(resourceExtensions);
+
+            var resourceType = ResourceGroupOperations.ResourceType;
+            foreach (var type in FindAllContainers())
+            {
+                var obj = Activator.CreateInstance(resourceExtensions.Assembly.FullName, type.Name, true,
+                    BindingFlags.CreateInstance | BindingFlags.NonPublic, null, null, CultureInfo.CurrentCulture, null);
+                var validResourceType = obj.GetType().GetProperty("ValidResourceType").GetValue(obj) as string;
+                if (resourceType.Equals(validResourceType))
+                {
+                    var getContainerMethod = resourceExtensions.GetMethod($"Get{type.Name.Substring(0, type.Name.IndexOf("Container"))}s");
+                    Assert.NotNull(getContainerMethod);
```

Comment:
Here you also need to check if the parent is `resourceGroups`.

Reply:
> var resourceType = ResourceGroupOperations.ResourceType;
> if (resourceType.Equals(validResourceType))
This should tell if a reource's parent is ResourceGroup

Reply:
Here you are only checking if the validResourceType are equal or not. You are not checking if the parent is `resourceGroups` or not.

Reply:
[SubscriptionOperations.ResourceType is "Microsoft.Resources/subscriptions"](https://github.com/Azure/azure-sdk-for-net/blob/7e6969d3858ca73fda4e9bd570502bf63f8a803e/sdk/resourcemanager/Azure.ResourceManager.Core/src/ResourceGroupContainer.cs#L35)
[ResourceGroupOperations.ResourceType is "Microsoft.Resources/subscriptions/resourceGroups"](https://github.com/Azure/azure-sdk-for-net/blob/7e6969d3858ca73fda4e9bd570502bf63f8a803e/sdk/resourcemanager/Proto.Client/compute/VirtualMachineContainer.cs#L41)
[VirtualNetworkOperations.ResourceType is "Microsoft.Network/virtualNetworks"](https://github.com/Azure/azure-sdk-for-net/blob/7e6969d3858ca73fda4e9bd570502bf63f8a803e/sdk/resourcemanager/Proto.Client/network/SubnetContainer.cs#L30)
By getting this value from ValidResourceType, we can simply tell if the parent is resourcegroup or not.

Reply:
Ahh, I see what you're doing here. 

Now you don't need to ignore this test case because ValidResourceType for container class is implemented. Check [here](https://github.com/Azure/autorest.csharp/blob/feature/v3/test/TestProjects/MgmtParent/Generated/AvailabilitySetContainer.cs#L21)

Reply:
You can refer [this](https://github.com/Azure/autorest.csharp/blob/2bae1226613297c34d48d53549796e18781b7c3c/test/AutoRest.TestServer.Tests/Mgmt/TestProjects/TestProjectTests.cs#L122-L126) example.